### PR TITLE
Add helm app registry support

### PR DESCRIPTION
**Pull request by joejulian**
Add support for the helm (app registry plugin)[https://github.com/app-registry/appr-helm-plugin] to allow version pinned charts to be used in k2 configs.

Depends on https://github.com/samsung-cnct/k2-tools/pull/84

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -310,18 +310,26 @@ You can try having helm install a new service, such as the Kubernetes dashboard
...
+
+```bash
 docker run $K2OPTS -e HELM_HOME=$HOME/.kraken/${CLUSTER}/.helm -e KUBECONFIG=$HOME/.kraken/${CLUSTER}/admin.kubeconfig quay.io/samsung_cnct/k2:latest helm search kubernetes-dashboard
 
 NAME                      	VERSION	DESCRIPTION                      
 atlas/kubernetes-dashboard	0.1.0  	A kubernetes dashboard Helm chart
 ```
 
-This indicates that the file to install is `atlas/kubernetes-dashboard`.
+This indicates that the file to install is `samsung_cnct/kubernetes-dashboard` from the quay.io registry.
```
Comment (DStorck) [inline README.md]: if it's the legacy repo, should that be atlas? if so, line 327 should also be atlas, otherwise line 324 should be samsung_cnct/

Comment (joejulian) [inline README.md]: Hmm, not what I meant at all. I'll fix that.

File: schemas/config/v1/helmChart.json [hunk 2]
```diff
@@ -26,10 +30,16 @@
...
     "version"
   ],
 
+  "oneOf": [
+      { "required":
+          [ "repo" ] },
+      { "required":
+          [ "registry" ] }
+  ],
+
```
Comment (joejulian) [inline schemas/config/v1/helmChart.json]: @davidewatson Here's the bit you were interested in seeing.

**Comment by leahnp** [comment]
I am also personally interested in the difference between repo and registry - also what is the reason for the switch to quay? Other than that LGTM

**Comment by joejulian** [comment]
imho, the biggest advantage is:

repo: you get what you get. If you're running a 1.4 cluster and a chart is upgraded for 1.6, it'll stop working for you.

registry: tagged versions. You can pin your charts.

**Comment by leahnp** [comment]
@joejulian do you know who is behind the registry project? Just wondering if we can rely on it long-term or if it is affiliated with k8s at all.

**Comment by DStorck** [comment]
@joejulian awesome - it'd be great to have that comment about tagged versions in the docs. Otherwise it's not clear immediately why you would use one over the other.

**Comment by venezia** [comment]
Oh another reason is private helm charts. Right now helm assumes charts are available with no authentication. There may be reasons for a chart to be private. App registry allows logins (quay logins for example)

Our configuration may also need to have surname and password support for a given app registry repository. 

Mike

**Comment by coffeepac** [comment]
to answer @DStorck's last comment:  https://github.com/samsung-cnct/k2/issues/629

**Pull request merged**